timesketch/ui/graphs/cytoscape.component.spec.ts

```diff
@@ -0,0 +1,264 @@
+
+import {CytoscapeComponent} from './cytoscape.component'
+
+export const testCytoscapeOptions = {
+  zoomingEnabled: false,
+  userZoomingEnabled: false,
+  panningEnabled: false,
+  userPanningEnabled: false,
+  boxSelectionEnabled: true,
+  autolock: true,
+  autoungrabify: true,
+  autounselectify: true,
+  headless: true,
+  hideEdgesOnViewport: true,
+  hideLabelsOnViewport: true,
+  textureOnViewport: true,
+  motionBlur: true,
+  styleEnabled: false,
+  minZoom: 1e-10,
+  maxZoom: 1e10,
+  touchTapThreshold: 10,
+  desktopTapThreshold: 6,
+  motionBlurOpacity: 0.4,
+  wheelSensitivity: 0.8,
+  pixelRatio: 1.0,
+  selectionType: 'additive',
+}
+
+export const getOptionFromCytoscape = (_cy: Cy.Core, option: string) => {
+  const cy = _cy as any
+  if (option in cy)
+    return cy[option]()
+  else if (option in cy.json())
+    return cy.json()[option]
+  else
+    return cy._private.options[option] || cy._private.renderer.options[option]
+}
+
+export const testCytoscapeEvents = [
+  'mousedown', 'mouseup', 'click', 'mouseover', 'mouseout', 'mousemove',
+  'touchstart', 'touchmove', 'touchend', 'tapstart', 'vmousedown',
+  'tapdrag', 'vmousemove', 'tapdragover', 'tapdragout', 'tapend',
+  'vmouseup', 'tap', 'vclick', 'taphold', 'cxttapstart', 'cxttapend',
+  'cxttap', 'cxtdrag', 'cxtdragover', 'cxtdragout', 'boxstart', 'boxend',
+  'boxselect', 'box', 'layoutstart', 'layoutready', 'layoutstop', 'ready',
+  'destroy', 'render', 'resize',
+]
+
+describe('CytoscapeComponent', () => {
+  beforeAll(() => {
+    TestBed.resetTestEnvironment()
+    TestBed.initTestEnvironment(
+      BrowserDynamicTestingModule, platformBrowserDynamicTesting(),
+    )
+  })
+
+  describe('options', () => {
+    let fixture: ComponentFixture<OptionsTestComponent>
+    beforeEach(async () => {
+      TestBed.configureTestingModule({
+        imports: [BrowserModule],
+        declarations: [CytoscapeComponent, OptionsTestComponent],
+      })
+      fixture = TestBed.createComponent(OptionsTestComponent)
+      // prevent 'Error: Timeout - Async callback was not invoked within timeout':
+      fixture.componentInstance.cytoscapeComponent['container'] = undefined
+      await fixture.whenStable()
+    })
+
+    for (const [option, value] of Object.entries(testCytoscapeOptions)) {
+      it(`should handle changes of "${option}" correctly`, () => {
+        fixture.detectChanges()
+        fixture.componentInstance.options[option] = value
+        fixture.detectChanges()
+        const cy = fixture.componentInstance.cytoscapeComponent.cy
+        const newValue = getOptionFromCytoscape(cy, option)
+        expect(newValue).toEqual(value)
+      })
+    }
+
+    for (const option_to_change of Object.keys(testCytoscapeOptions)) {
+      it(`should retain other options after changing "${option_to_change}"`, () => {
+        fixture.componentInstance.options = {...testCytoscapeOptions}
+        fixture.detectChanges()
+        if (option_to_change === 'styleEnabled')
```

Comment:
Curly braces here again (and through-out the CL).

Reply:
Sure, I can add a rule to the linter.

Reply:
Done.